### PR TITLE
Added A Dashboard section and made a demo section according to it .

### PR DESCRIPTION
**Pull request by AmanSal1**
PULL REQUEST DESCRIPTION :

Added the 'Dashboard' section and then included the 'Demo' section within it. This section contains different tabs for various flow status, where each flow is assigned to its designated tab only. Mock data for this section is being generated using the "FlowRunMocks" function and then passed on to the different tabs .

### CODE REVIEW

**Comment by pleek91** [comment]
@AmanSal1 thanks for picking this up! Couple minor things to note

1. There's a component for this called `FlowRunStateTypeTabs.vue`. 
2. I'd expect a separate demo for that component rather than adding to the `DashboardTimeSpanFilter` demo
3. Looks like there are some typescript errors

**Comment by pleek91** [comment]
> @pleek91 I have a doubt then the content of DashboardTimeSpanFilter demo would be similar to FlowRunStateTypeTabs.vue or it should contain something else ?

@AmanSal1 generally our demo sections are 1-1 with a component. So I'd only expect to see one component in each demo. So the time span demo should be reverted and a new demo section would be needed for the `FlowRunStateTypeTabs`. That demo would actually use that component and demo how it is used rather than copying the contents of it. Hopefully that makes sense! If not let me know and I'll try to clarify

**Comment by AmanSal1** [comment]
Oh okay got some clarity but then what would be the demo section named for the FlowRunStateTypeTabs  ?

**Comment by AmanSal1** [comment]
Oh okay . I have made the interface with the FlowRunStateTypeTab but I am unable to show the data in each tab according to their state . I think that we have to use the prop but I am not able to figure it out .

**Comment by AmanSal1** [comment]
@pleek91 Can you review the logic bit stuck with adding the mock data in each tab according to their state.

File: demo/sections/dashboard/DashboardTimeSpanFilter.vue
Comment (pleek91) [inline demo/sections/dashboard/DashboardTimeSpanFilter.vue:46]: @AmanSal1 to get some data to show up try removing all this and just adding the `useFlowRunsMock(7)`. The `FlowRunsStateTypeTab` component should automatically get data from the mocked api. If that doesn't work I'll clone your branch and figure out what's going on. 
```suggestion
 useFlowRunsMock(7)
```

File: demo/sections/dashboard/DashboardTimeSpanFilter.vue
Comment (pleek91) [inline demo/sections/dashboard/DashboardTimeSpanFilter.vue:1]: This component should be `dashboard/FlowRunStateTypeTab.vue` and not `dashboard/DashboardTimeSpanFilter.vue`

Comment (AmanSal1) [reply demo/sections/dashboard/DashboardTimeSpanFilter.vue:1]: Made the changed as you said can you review it now ?

Comment (AmanSal1) [reply demo/sections/dashboard/DashboardTimeSpanFilter.vue:1]: @pleek91 I guess there are some linting issue but can you tell me is the code correct now and then in another commit I would fix the linting of the code

**Comment by pleek91** [comment]
@AmanSal1 the demo looks correct to me. Though its still not loading data the way I would expect. But I can sort that out later. You will need to remove the `DashboardTimeSpanFilter.vue` demo component that isn't used now though and fix the linting issues). 

I'm not sure how adding this demo works with fixing https://github.com/PrefectHQ/prefect-ui-library/issues/1579 though as the actual component itself hasn't change any.

**Comment by AmanSal1** [comment]
@pleek91 
1.In the second commit, I tried writing the logic for it because just using the useflowrunmock was not showing the data as it should be, like separately in each tab.
2. I don't know why it's showing "DashboardTimeSpanFilter.vue" when I have renamed it. I will check it again. 
3. So basically, there is some confusion regarding this issue #1579, which requires cloud access that I guess I don't have. Therefore, the PR I have raised is for adding the missing demo section in the Prefect UI library's current version.
4. Additionally, could you please provide the command for linting? That would be great

**Comment by pleek91** [comment]
> @pleek91 1.In the second commit, I tried writing the logic for it because just using the useflowrunmock was not showing the data as it should be, like separately in each tab. 2. I don't know why it's showing "DashboardTimeSpanFilter.vue" when I have renamed it. I will check it again. 3. So basically, there is some confusion regarding this issue #1579, which requires cloud access that I guess I don't have. Therefore, the PR I have raised is for adding the missing demo section in the Prefect UI library's current version. 4. Additionally, could you please provide the command for linting? That would be great

1-2. Understood
3. Adding the demo is great. We should just unlink this from that issue so it doesn't get closed when this gets merged. 
4. `npm run lint:fix`. To see all the available commands check the [package.json scripts](https://github.com/PrefectHQ/prefect-ui-library/blob/main/package.json#L17) section

**Comment by AmanSal1** [comment]
So, should I create a new issue or what should be done according to you?

**Comment by AmanSal1** [comment]
@pleek91 Or maybe I should remove the fixes keyword so it does closes the issue or I guess their is some feature of not closing the issue when a PR is merged but then in both the scenarios it would be difficult to keep a track of what has been added . 
Its on your call whatever you recommend

**Comment by pleek91** [comment]
@AmanSal1 I updated the description to remove the fixes keyword which unlinked the PR 👍

**Comment by AmanSal1** [comment]
@AmanSal1 Done with the linting you can review now it  and I guess now you can see the FlowRunStateTypeTab component

**Comment by AmanSal1** [comment]
@pleek91 Hey there! Whenever you find some free time, would you mind taking a moment to review it?

**Review by znicholasbrown — APPROVED** [review]
Thanks @AmanSal1 - this LGTM, feel free to merge.

**Comment by AmanSal1** [comment]
@znicholasbrown waiting for the approval from @pleek91

**Comment by znicholasbrown** [comment]
They're out of office for the next couple of weeks so you don't need to wait for their approval

**Comment by AmanSal1** [comment]
@znicholasbrown Then  how will i merge this PR ?

**Comment by znicholasbrown** [comment]
Sorry @AmanSal1, I didn't realize the pending review was blocking your ability to merge - if it's alright with you I'll go ahead and merge on your behalf

**Comment by AmanSal1** [comment]
@znicholasbrown So, will you be pressing the merge button, or will you be cloning these changes and then publishing them on your behalf??

**Comment by znicholasbrown** [comment]
I'll just press the button, the commits will still be yours!

**Comment by AmanSal1** [comment]
@znicholasbrown oh that’s totally fine I thought i have to share my credentials to you 😅sorry

**Comment by znicholasbrown** [comment]
All good! I'll go ahead and merge this then

**Comment by AmanSal1** [comment]
Yes go ahead !! Sorry for the misunderstanding though

**Pull request merged**
